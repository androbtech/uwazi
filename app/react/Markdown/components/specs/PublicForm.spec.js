import React from 'react';
import { shallow } from 'enzyme';
import Immutable from 'immutable';
import { LocalForm } from 'react-redux-form';
import PublicForm from '../PublicForm.js';

describe('PublicForm', () => {
  let props;
  let component;
  let instance;
  let submit;
  let request;

  beforeEach(() => {
    request = Promise.resolve({ promise: Promise.resolve('ok') });
    submit = jasmine.createSpy('submit').and.returnValue(request);
  });

  const render = (customProps) => {
    props = {
      template: Immutable.fromJS({ _id: '123', properties: [{ type: 'text', name: 'text' }] }),
      thesauris: Immutable.fromJS([]),
      file: false,
      attachments: false,
      submit,
      remote: false,
    };
    const mappedProps = { ...props, ...customProps };
    component = shallow(<PublicForm.WrappedComponent {...mappedProps}/>);
    instance = component.instance();
    instance.refreshCaptcha = jasmine.createSpy('refreshCaptcha');
    instance.formDispatch = jasmine.createSpy('formDispatch');
  };

  it('should render a form', () => {
    render();
    expect(component).toMatchSnapshot();
  });

  it('should render a form with file and attachments', () => {
    render({ file: true, attachments: true });
    expect(component).toMatchSnapshot();
  });

  it('should submit the values', () => {
    render();
    component.find(LocalForm).simulate('submit', { title: 'test' });
    expect(props.submit).toHaveBeenCalledWith({ file: undefined, title: 'test', template: '123' }, false);
  });

  it('should refresh the captcha and clear the form after submit', (done) => {
    render();
    component.find(LocalForm).simulate('submit', { title: 'test' });
    request.then((uploadCompletePromise) => {
      uploadCompletePromise.promise.then(() => {
        expect(instance.formDispatch).toHaveBeenCalledWith({ model: 'publicform', type: 'rrf/reset' });
        expect(instance.refreshCaptcha).toHaveBeenCalled();
        done();
      });
    });
  });
});
