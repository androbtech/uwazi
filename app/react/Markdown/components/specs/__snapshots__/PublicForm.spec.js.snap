// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicForm should render a form 1`] = `
<div
  className="public-form"
>
  <LocalForm
    getDispatch={[Function]}
    initialState={Object {}}
    model="publicform"
    onSubmit={[Function]}
    validators={
      Object {
        "captcha": Object {
          "required": [Function],
        },
        "title": Object {
          "required": [Function],
        },
      }
    }
  >
    <Connect(FormGroup)
      key="title"
      model=".title"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label
            htmlFor="title"
          >
            <Connect(Translate)>
              Title
            </Connect(Translate)>
            <span
              className="required"
            >
              *
            </span>
          </label>
        </li>
        <li
          className="wide"
        >
          <DefaultConnectedControlText
            className="form-control"
            id="title"
            model=".title"
          />
        </li>
      </ul>
    </Connect(FormGroup)>
    <Connect(MetadataFormFields)
      model="publicform"
      template={
        Immutable.Map {
          "_id": "123",
          "properties": Immutable.List [
            Immutable.Map {
              "type": "text",
              "name": "text",
            },
          ],
        }
      }
      thesauris={Immutable.List []}
    />
    <Connect(FormGroup)
      key="captcha"
      model=".captcha"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label>
            <Connect(Translate)>
              Captcha
            </Connect(Translate)>
            <span
              className="required"
            >
              *
            </span>
          </label>
        </li>
        <li
          className="wide"
        >
          <Captcha
            model=".captcha"
            refresh={[Function]}
            remote={false}
          />
        </li>
      </ul>
    </Connect(FormGroup)>
    <input
      className="btn btn-success"
      type="submit"
      value="Submit"
    />
  </LocalForm>
</div>
`;

exports[`PublicForm should render a form with file and attachments 1`] = `
<div
  className="public-form"
>
  <LocalForm
    getDispatch={[Function]}
    initialState={Object {}}
    model="publicform"
    onSubmit={[Function]}
    validators={
      Object {
        "captcha": Object {
          "required": [Function],
        },
        "title": Object {
          "required": [Function],
        },
      }
    }
  >
    <Connect(FormGroup)
      key="title"
      model=".title"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label
            htmlFor="title"
          >
            <Connect(Translate)>
              Title
            </Connect(Translate)>
            <span
              className="required"
            >
              *
            </span>
          </label>
        </li>
        <li
          className="wide"
        >
          <DefaultConnectedControlText
            className="form-control"
            id="title"
            model=".title"
          />
        </li>
      </ul>
    </Connect(FormGroup)>
    <Connect(MetadataFormFields)
      model="publicform"
      template={
        Immutable.Map {
          "_id": "123",
          "properties": Immutable.List [
            Immutable.Map {
              "type": "text",
              "name": "text",
            },
          ],
        }
      }
      thesauris={Immutable.List []}
    />
    <div
      className="form-group"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label
            htmlFor="file"
          >
            <Connect(Translate)>
              Document
            </Connect(Translate)>
            <DefaultConnectedControlFile
              accept=".pdf"
              className="form-control"
              id="file"
              model=".file"
            />
          </label>
        </li>
      </ul>
    </div>
    <div
      className="form-group"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label
            htmlFor="attachments"
          >
            <Connect(Translate)>
              Attachments
            </Connect(Translate)>
            <DefaultConnectedControlFile
              className="form-control"
              id="attachments"
              model=".attachments"
              multiple="multiple"
            />
          </label>
        </li>
      </ul>
    </div>
    <Connect(FormGroup)
      key="captcha"
      model=".captcha"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label>
            <Connect(Translate)>
              Captcha
            </Connect(Translate)>
            <span
              className="required"
            >
              *
            </span>
          </label>
        </li>
        <li
          className="wide"
        >
          <Captcha
            model=".captcha"
            refresh={[Function]}
            remote={false}
          />
        </li>
      </ul>
    </Connect(FormGroup)>
    <input
      className="btn btn-success"
      type="submit"
      value="Submit"
    />
  </LocalForm>
</div>
`;
